services/ext/api.go

```diff
@@ -581,6 +582,27 @@ func (api *PublicAPI) CanceledRequestsToJoinForCommunity(id types.HexBytes) ([]*
 	return api.service.messenger.CanceledRequestsToJoinForCommunity(id)
 }
 
+// Generates a single hash for each address that needs to be revealed to a community.
+// Each hash needs to be signed.
+// The order of retuned hashes corresponds to the order of addresses in addressesToReveal.
+func (api *PublicAPI) GenerateJoiningCommunityRequestsForSigning(memberPubKey string, communityID types.HexBytes, addressesToReveal []string) ([]account.SignParams, error) {
+	return api.service.messenger.GenerateJoiningCommunityRequestsForSigning(memberPubKey, communityID, addressesToReveal)
+}
+
+// Generates a single hash for each address that needs to be revealed to a community.
+// Each hash needs to be signed.
+// The order of retuned hashes corresponds to the order of addresses in addressesToReveal.
+func (api *PublicAPI) GenerateEditCommunityRequestsForSigning(memberPubKey string, addressesToReveal []string) ([]account.SignParams, error) {
```

Comment:
I don't quite understand what is this API for?

Reply:
What exactly confuses you? There are 2 types of requests, one is generated when the user is joining a community, and another one, edit request, is generated when the user wants to edit/change addresses that he previously revealed to the community.

Reply:
I had some difficulty grasping `GenerateEditCommunityRequestsForSigning` when it propagated an empty community ID. Now that it's been corrected, it's clear.